### PR TITLE
Fix docs of is_diagonalizable

### PR DESCRIPTION
**Pull request by sylee957**
#### Brief description of what is fixed or changed

I think that I have misused the notations like `orthogonally_diagonalizable` in #16778 
And `reals_only` keyword seems like defined with whether all the eigenvalues are real or not.
So, in this case, complex hermitian matrix can also be considered diagonalizable as they have real eigenvalues.
(Even though they can possibly return `P` matrix with complex entries)

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/matrices/eigen.py [hunk 1]
```diff
@@ -322,10 +322,10 @@ def _is_diagonalizable(M, reals_only=False, dotprodsimp=None, **kwargs):
 
     reals_only : bool, optional
         If ``True``, it tests whether the matrix can be diagonalized
-        without complex numbers. (Orthogonally diagonalizable)
+        with a diagonal matrix without complex numbers.
```
Comment (smichr) [inline sympy/matrices/eigen.py]:
```suggestion
        to contain only real numbers on the diagonal.

```

File: sympy/matrices/eigen.py [hunk 2]
```diff
@@ -322,10 +322,10 @@ def _is_diagonalizable(M, reals_only=False, dotprodsimp=None, **kwargs):
 
     reals_only : bool, optional
         If ``True``, it tests whether the matrix can be diagonalized
-        without complex numbers. (Orthogonally diagonalizable)
+        with a diagonal matrix without complex numbers.
 
-        If ``False``, it tests whether the matrix can be unitarily
-        diagonalizable.
+        If ``False``, it tests whether the matrix can be diagonalized
+        with a diagonal matrix with complex numbers.
```
Comment (smichr) [inline sympy/matrices/eigen.py]:
```suggestion
        at all, even with numbers that may not be real.

```

File: sympy/matrices/eigen.py [hunk 3]
```diff
@@ -348,8 +348,8 @@ def _is_diagonalizable(M, reals_only=False, dotprodsimp=None, **kwargs):
     >>> M.is_diagonalizable()
     False
 
-    Example of a unitarily diagonalizable, but not orthogonally
-    diagonalizable:
+    Example of a matrix which is diagonalizable with a diagonal matrix
```
Comment (smichr) [inline sympy/matrices/eigen.py]:
```suggestion
    Example of a matrix that is diagonalized in terms of non-real entries:

```

File: sympy/matrices/eigen.py [hunk 4]
```diff
@@ -384,8 +384,7 @@ def _is_diagonalizable(M, reals_only=False, dotprodsimp=None, **kwargs):
     if all(e.is_real for e in M) and M.is_symmetric():
         return True
 
-    if all(e.is_complex for e in M) and M.is_hermitian \
-            and not reals_only:
+    if all(e.is_complex for e in M) and M.is_hermitian:
```
Comment (smichr) [inline sympy/matrices/eigen.py]: I'm not sure if this is covered...the `is_diagonalizable` tests are pretty sparse. Something like
```assert Matrix([[0,-4*I],[4*I,0]]).is_diagonalizable()` would work. And for a non-Hermitian case that is diagonalizable in terms of reals, `assert Matrix([[0,-3*I],[4*I,0]]).is_diagonalizable(reals_only=True)`

File: sympy/matrices/eigen.py [hunk 5]
```diff
@@ -348,8 +350,9 @@ def _is_diagonalizable(M, reals_only=False, dotprodsimp=None, **kwargs):
     >>> M.is_diagonalizable()
     False
 
-    Example of a unitarily diagonalizable, but not orthogonally
-    diagonalizable:
+    Example of a matrix that is diagonalized in terms of non-real entries:
+
+    with complex entries, but not with real entries:
```
Comment (smichr) [inline sympy/matrices/eigen.py]:
```suggestion
```

File: sympy/matrices/eigen.py (hunk 4 above)
Comment (sylee957) [reply sympy/matrices/eigen.py]: But I'm afraid that such lemmas like
```python
x, y = symbols('x y', real=True)
Matrix([[0, x*I], [y*I, 0]]).is_diagonalizable
```
can be too trivial.

File: sympy/matrices/eigen.py (hunk 5 above)
Comment (smichr) [reply sympy/matrices/eigen.py]: this line needs to be deleted

File: sympy/matrices/eigen.py (hunk 4 above)
Comment (smichr) [reply sympy/matrices/eigen.py]: At least such tests would provide nominal coverage. I see, however, that codecov indicates that these lines are covered. So once the redundant line is deleted this should be ready to go.

File: sympy/matrices/eigen.py [hunk 6]
```diff
@@ -348,8 +350,8 @@ def _is_diagonalizable(M, reals_only=False, dotprodsimp=None, **kwargs):
     >>> M.is_diagonalizable()
     False
 
-    Example of a unitarily diagonalizable, but not orthogonally
-    diagonalizable:
+    Example of a matrix that is diagonalized in terms of non-real entries:
+
```
Comment (sylee957) [inline sympy/matrices/eigen.py]:
```suggestion

```

**Pull request merged**